### PR TITLE
Fix broken `eqErrGRPC` helper function in integration tests

### PR DESCRIPTION
**Pull request by zhijun42**
The test helper function `eqErrGRPC` has always been flawed since introduced in 2016. 

```go
func eqErrGRPC(err1 error, err2 error) bool {
	return !(err1 == nil && err2 != nil) || err1.Error() == err2.Error()
}
```

When the first argument was non-nil (usually the case in tests), it will always return true. This has made all error assertions using this helper effectively no-ops.

**Changes:**
- Replace broken implementation with correct gRPC status comparison using `status.FromError()`
- Handle `rpctypes.EtcdError` so `eqErrGRPC` works in grpcproxy mode where `ToGRPC` adapters bypass the gRPC wire
- Handle nil errors (both nil → equal, one nil → not equal)
- Add unit tests covering code/message permutations and edge cases

**Pre-existing test bugs exposed by the fix:**
- `TestV3PutIgnoreValue`: sent `IgnoreValue=true` with a non-empty `Value` field, causing `ErrGRPCValueProvided` instead of the expected `ErrGRPCKeyNotFound`. Fixed by clearing `Value`.
- `TestV3TooLargeRequest`: sent a 2MB value which exceeded the gRPC `MaxRecvMsgSize` (`MaxRequestBytes` 1.5MB + 512KB overhead = 2MB), causing gRPC `ResourceExhausted` instead of etcd's `ErrGRPCRequestTooLarge`. Fixed by reducing the value size to stay under the gRPC transport limit.
- `TestV3LeaseRecoverKeyWithMultipleLease`: after server restart, `leaseExist()` used the old client which hadn't reconnected yet. The broken `eqErrGRPC` masked the connection error by returning true. Fixed by adding `WaitStarted` to ensure client connectivity before checking leases.
- `TestV3LargeRequests`: skipped in grpcproxy mode as the proxy does not propagate custom `MaxRequestBytes` to its internal gRPC client connection.

### CODE REVIEW

File: tests/integration/v3_grpc_test.go [hunk 1]
```diff
@@ -1961,8 +1961,122 @@ func TestV3AdditionalGRPCOptions(t *testing.T) {
...
+//	// Correct: compare gRPC errors (from raw gRPC client via ToGRPC)
+//	eqErrGRPC(err, rpctypes.ErrGRPCUserEmpty)
+//
+//	// Wrong: DO NOT mix client-side (ErrXxx) with server-side (ErrGRPCXxx) errors.
+//	// Client-side errors use a different type (EtcdError) that returns codes.Unknown.
+//	eqErrGRPC(rpctypes.ErrUserEmpty, rpctypes.ErrGRPCUserEmpty) // returns false!
+//
+//	// For errors from clientv3, use errors.Is() with client-side error types:
+//	errors.Is(err, rpctypes.ErrUserEmpty)
 func eqErrGRPC(err1 error, err2 error) bool {
```
Comment (serathius) [inline tests/integration/v3_grpc_test.go:2275]: This function is badly designed, it promises too much, validates too little. Instead of writing a essay on how to use a test function, just panic if user provides incorrect input.

Comment (zhijun42) [reply tests/integration/v3_grpc_test.go:2275]: Good point! Refactored to just panic. And removed my essay.

File: tests/integration/v3_grpc_test.go [hunk 2]
```diff
@@ -1961,8 +1961,133 @@ func TestV3AdditionalGRPCOptions(t *testing.T) {
 	}
 }
 
+// eqErrGRPC validates both inputs must be gRPC status errors
+// (e.g., rpctypes.ErrGRPCXxx) or nil, and then compares them.
```
Comment (serathius) [inline tests/integration/v3_grpc_test.go]: Instead of writing comments improve function name.

Comment (zhijun42) [reply tests/integration/v3_grpc_test.go]: I can't come up with a better name.

- matchGRPCErrors
- errorsEqualByGRPCStatus
- mustEqualGRPCErrors
- mustGRPCErrorsEqual
- verifyGRPCErrorsEqual
- requireGRPCErrorsEqual
- validateAndCompareGRPCErrors
- grpcStatusCodeAndMessageEqual

Comment (zhijun42) [reply tests/integration/v3_grpc_test.go]: Simply removed the comment since the function is very straightforward. Also removed unmeaningful test cases to further reduce the lines of change in this PR.

**Review by serathius — APPROVED** [review]

**Review by serathius — CHANGES_REQUESTED** [review]
Please fix tests

**Comment by serathius** [comment]
```
--- FAIL: TestV3CorruptAlarm (4.32s)
panic: eqErrGRPC: not a gRPC status error: rpctypes.EtcdError etcdserver: corrupt cluster [recovered, repanicked]
```

**Comment by zhijun42** [comment]
@serathius All tests pass now. As I was fixing the CI, I found many more pre-existing bugs masked by the broken `eqErrGRPC` helper. I fixed them all and recorded in the PR description.

**Comment by serathius** [comment]
Can you resolve merge conflict?

**Review by serathius — APPROVED** [review]

**Comment by serathius** [comment]
PTAL @ahrtr @fuweid

File: tests/integration/v3_alarm_test.go [hunk 3]
```diff
@@ -223,7 +224,7 @@ func TestV3CorruptAlarm(t *testing.T) {
 	for i := 0; i < 5; i++ {
 		presp, perr := clus.Client(0).Put(t.Context(), "abc", "aaa")
 		if perr != nil {
-			if eqErrGRPC(perr, rpctypes.ErrCorrupt) {
+			if errors.Is(perr, rpctypes.ErrCorrupt) {
```
Comment (ahrtr) [inline tests/integration/v3_alarm_test.go]: not sure why change it? `errors.Is` might be doing almost the same thing as the new implementation of `eqErrGRPC`, but not sure if there are any differences.

Comment (zhijun42) [reply tests/integration/v3_alarm_test.go]: Yeah good catch! You're right, it's not really needed. Cleaned these up.

**Comment by zhijun42** [comment]
I think this PR is good to go. Can we merge it?

**Review by ahrtr — APPROVED** [review]
